src/common/contracts.py

```diff
@@ -92,6 +95,36 @@ async def get_validators_index(self) -> int:
         """Fetches vault's current validators index."""
         return await self.contract.functions.validatorIndex().call()
 
+    async def get_registered_validators_public_keys(self) -> list[HexStr]:
+        """Fetches the validator registered events."""
+        events = await self._get_events(
+            self.events.ValidatorRegistered,
+            from_block=settings.network_config.KEEPER_GENESIS_BLOCK,
+            to_block=await execution_client.eth.get_block_number(),
+        )
+        return [Web3.to_hex(event['args']['publicKey']) for event in events]
+
+
+class V2PoolContract(ContractWrapper):
+    abi_path = 'abi/IV2Pool.json'
+    settings_key = 'V2_POOL_CONTRACT_ADDRESS'
+
+    async def get_registered_validators_public_keys(self) -> list[HexStr]:
+        """Fetches the validator registered events."""
+        blocks_per_month = int(SECONDS_PER_MONTH // settings.network_config.SECONDS_PER_BLOCK)
+        to_block = BlockNumber(
+            min(
+                settings.network_config.KEEPER_GENESIS_BLOCK + blocks_per_month,
+                await execution_client.eth.get_block_number(),
+            )
+        )
```

Comment:
Business logic should be in `_fetch_registered_validators`, not here. Contract  method should just take params as is or use simple defaults.